### PR TITLE
Skip validation

### PR DESCRIPTION
**Pull request by suzhoum**
*Issue #, if available:*
By default, a final validation is run after the training is signaled to stop, and as a result, AutoPilot team's 10min quick build on large dataset overshoots by quite a long time.

*Description of changes:*
By passing in a `skip_final_val` flag, we disable the final validation by setting the `trainer.val_dataloaders = []` to stop the validation.

By setting:
```
hyperparameters = {
    "optimization.top_k_average_method": "best",
    "optimization.val_check_interval": 1.0,
    "optimization.skip_final_val" = True,
}
```
we were able to reduce the `.fit()` time from 30-40min down to 10min.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/tests/unittests/others_2/test_predictor_advanced.py [hunk 1]
```diff
@@ -81,3 +81,26 @@ def test_predictor_gradient_checkpointing(
...
+    save_path = "petfinder_checkpoint"
+
+    predictor = MultiModalPredictor(
+        label=dataset.label_columns[0], problem_type=dataset.problem_type, eval_metric=dataset.metric, path=save_path
+    )
+    hyperparameters = {
+        "model.names": ["timm_image", "hf_text", "fusion_mlp"],
+        "model.timm_image.checkpoint_name": "ghostnet_100",
+        "model.hf_text.checkpoint_name": "nlpaueb/legal-bert-small-uncased",
+        "optimization.skip_final_val": True,
```
Comment (zhiqiangdon) [inline multimodal/tests/unittests/others_2/test_predictor_advanced.py:97]: Better to add

```
"env.num_workers": 0,
"env.num_workers_evaluation": 0,
```

File: docs/tutorials/multimodal/advanced_topics/customization.md [hunk 2]
```diff
@@ -256,6 +256,17 @@ predictor.fit(hyperparameters={"optimization.efficient_finetune": "bit_fit"})
...
 
+### optimization.skip_final_val
+
+Whether to skip the final validation after training is signaled to stop.
+
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"optimization.skip_final_val": False})
+# skip the final validation
+predictor.fit(hyperparameters={"optimization.top_k_average_method": True})
```
Comment (bryanyzhu) [inline docs/tutorials/multimodal/advanced_topics/customization.md:267]: Maybe a typo here? I think it should be "optimization.skip_final_val": True

**Review by bryanyzhu — COMMENTED** [review]
Thanks for the quick fix.

File: multimodal/src/autogluon/multimodal/predictor.py [hunk 3]
```diff
@@ -1631,6 +1632,7 @@ def _default_predict(
                 model_postprocess_fn=self._model_postprocess_fn,
                 efficient_finetune=OmegaConf.select(self._config, "optimization.efficient_finetune"),
                 trainable_param_names=trainable_param_names,
+                skip_final_val=self._config.optimization.skip_final_val,
```
Comment (zhiqiangdon) [inline multimodal/src/autogluon/multimodal/predictor.py:1635]: The validation is not called during inference. Maybe no need to set this flag.

Comment (suzhoum) [reply multimodal/src/autogluon/multimodal/predictor.py]: Good point! Removed it.

File: multimodal/src/autogluon/multimodal/optimization/lit_module.py [hunk 4]
```diff
@@ -233,6 +235,17 @@ def training_step(self, batch, batch_idx):
...
 
+    def on_validation_start(self) -> None:
+        if self.skip_final_val and self.trainer.should_stop:
+            self.trainer.val_dataloaders = []  # skip the final validation by setting val_dataloaders empty
+            self.log(
+                self.validation_metric_name,
+                self.validation_metric,
+                on_step=False,
+                on_epoch=True,
+            )
```
Comment (zhiqiangdon) [inline multimodal/src/autogluon/multimodal/optimization/lit_module.py:246]: What is this log used for?

**Review by zhiqiangdon — APPROVED** [review]
LGTM!

**Review by bryanyzhu — APPROVED** [review]
LGTM

**Pull request merged**